### PR TITLE
Refactored pymc3_random_discrete to use histograms instead of raw counts

### LINKED ISSUE #4810: `test_distributions_random` failing due to change in `st.chisquare` behavior
The tests in `test_distributions_random` which rely on `pymc3_random_discrete` started failing in the `v3` branch due to changes in the behavior of `st.chisquare`. I don't think any test is making use of this helper in the `main` branch after the `v4` refactoring, so we are not seeing failures related to that here.

For instance, in the `v3` branch, `test_distributions_random.py::TestScalarParameterSamples::test_poisson` passes locally with scipy `1.6.3` but fails with `1.7.0`

It raises a `ValueError` in `st.chisquare`:

```python
            if diff_gt_tol:
                msg = (f"For each axis slice, the sum of the observed "
                       f"frequencies must agree with the sum of the "
                       f"expected frequencies to a relative tolerance "
                       f"of {rtol}, but the percent differences are:\n"
                       f"{relative_diff}")
>               raise ValueError(msg)
E               ValueError: For each axis slice, the sum of the observed frequencies must agree with the sum of the expected frequencies to a relative tolerance of 1e-08, but the percent differences are:
E               0.006036217303822937
```
This issue / breaking change was added in https://github.com/scipy/scipy/pull/13193 and mentioned in https://github.com/scipy/scipy/issues/13362 and https://github.com/scipy/scipy/issues/14298

It seems like we need to normalize the inputs as the behavior of `st.chisquare` changed.

_Originally posted by @ricardoV94 in https://github.com/pymc-devs/pymc3/pull/4808#issuecomment-869192843_

### PR DESCRIPTION
**Pull request by fonnesbeck**
Fixes failures due to new error checking in `scipy.stats.chisquare`.

Closes #4810

### CODE REVIEW

File: pymc3/tests/test_distributions_random.py [hunk 1]
```diff
@@ -107,15 +107,12 @@ def pymc3_random_discrete(
...
-                expected[e] = (observed.get(e, 0), expected[e])
-            k = np.array([v for v in expected.values()])
-            if np.all(k[:, 0] == k[:, 1]):
+            observed, _ = np.histogram(o, bins=min(7, len(set(o))))
+            expected, _ = np.histogram(e, bins=min(7, len(set(o))))
+            if np.all(observed == expected):
                 p = 1.0
             else:
-                _, p = st.chisquare(k[:, 0], k[:, 1])
+                _, p = st.chisquare(observed+1, expected+1)
```
Comment (ricardoV94) [inline pymc3/tests/test_distributions_random.py]: What does the `+1` achieve?

Comment (fonnesbeck) [reply pymc3/tests/test_distributions_random.py]: If any of the bins are zero, the test fails (divide by zero). This was an easier solution than trying to te-generate the histogram with different bin sizes.

**Review by ricardoV94 — COMMENTED** [review]
LGTM, just needs that pre-commit fixed

**Review by michaelosthege — DISMISSED** [review]

**Review by michaelosthege — APPROVED** [review]

**Pull request merged**